src/instructlab/configuration.py

```diff
@@ -268,6 +268,23 @@ def api_base(self):
         return get_api_base(self.server.host, self.server.port)
 
 
+class _convert(BaseModel):
+    """Class describing configuration of the 'ilab rag convert' sub-command."""
+
+    output_dir: StrictStr = Field(
+        default_factory=lambda: DEFAULTS.CONVERTED_DOCUMENTS_DIR,
+        description="Directory where converted documents are stored.",
+    )
+    taxonomy_path: StrictStr = Field(
+        default_factory=lambda: DEFAULTS.TAXONOMY_DIR,
+        description="Directory where taxonomy is stored and accessed from.",
+    )
+    taxonomy_base: StrictStr = Field(
+        default=DEFAULTS.TAXONOMY_BASE,
+        description="Branch of taxonomy used to calculate diff against.",
```

Comment:
Why are the first two options using `default_factory` and `lambda` expressions but the third is not?

Not saying this is incorrect, just trying to understand the inconsistency here

Reply:
It is this way for here because we followed what is being done in `_generate` for pulling the taxonomy_path and taxonomy_base.  More broadly, I think lambda is consistently used throughout configuration.py for anything involving directories where things are stored and pulled from -- and often not used for other things.  That probably makes sense abstractly because directory structure seems like the sort of thing that could be dynamic in the future even if it is not now.

Reply:
IIUC, `default_factory` is especially useful when we want to mock the current data in test functions. For instance this is used to generate an envinronment-agnostic `default_config` with the `make regenerate-testdata` target. 
```py
@patch.multiple(
    config.DEFAULTS,
    _cache_home="/cache/instructlab",
    _data_dir="/data/instructlab",
)
```

Reply:
Makes sense - thanks for the explanation!